pages/instructorAssessmentInstances/instructorAssessmentInstances.ejs

```diff
@@ -129,29 +212,63 @@
+                         tabindex="0" data-toggle="popover" data-trigger="manual" data-container="body"
+                         data-html="true" data-placement="auto" title="Set Time Limit"
+                         data-content="
+                                       <p>
+                                         Time limit: <%= row.total_time %><br/>
+                                         Remaining time: <%= row.time_remaining %>
+                                       </p>
+                                       <form name=&quot;set-time-limit-form&quot; method=&quot;POST&quot;>
+                                         <input type=&quot;hidden&quot; name=&quot;__action&quot; value=&quot;set_time_limit&quot;>
+                                         <input type=&quot;hidden&quot; name=&quot;__csrf_token&quot; value=&quot;<%= __csrf_token %>&quot;>
+                                         <input type=&quot;hidden&quot; name=&quot;assessment_instance_id&quot; value=&quot;<%= row.assessment_instance_id %>&quot;>
+                                         <select class=&quot;form-control select-time-limit&quot; name=plus_minus>
+                                         <% if (row.time_remaining != 'Unlimited') { %>
+                                           <option value=+1>Add</option>
+                                           <option value=-1>Subtract</option>
+                                         <% } %>
+                                           <option value=set_total>Set time limit to</option>
+                                           <option value=set_rem>Set remaining time to</option>
+                                         <% if (row.time_remaining != 'Unlimited') { %>
+                                           <option value=unlimited>Set as unlimited</option>
+                                         <% } %>
+                                         </select>
+                                         <p class=&quot;form-inline&quot;>
+                                           <input class=&quot;form-control time-limit-field&quot; type=&quot;number&quot; name=&quot;time_add&quot; style=&quot;width: 5em&quot; value=&quot;5&quot;>
+                                           <select class=&quot;form-control time-limit-field&quot; name=time_ref><option value=60>minutes</option><option value=1>seconds</option></select>
+                                         </p>
+                                         <button type=&quot;button&quot; class=&quot;btn btn-secondary&quot; onclick=&quot;$('#row<%= iRow %>PopoverSetTimeLimit').popover('hide')&quot;>
+                                           Cancel
+                                         </button>
+                                         <button type=&quot;submit&quot; class=&quot;btn btn-success&quot;>Set</button>
+                                       </form>
+                                       ">
+                    </div>
```

Comment:
Rather than putting the "Set time limit" in "Actions", could we instead put an ["edit" button](https://github.com/PrairieLearn/PrairieLearn/blob/master/pages/instructorAssessmentInstance/instructorAssessmentInstance.ejs#L64) next to the current time in the "Remaining" column? I like generally having the "edit" functionality directly connected to the thing being edited.

Given that the "Remaining" column is were we currently put "Open/Closed" (and where we will put "Expired" or whatever once #3622 is implemented), I'm thinking that it would be more logical to put all of the close/re-open/timer functionality into an edit button within the "Remaining" column. This edit button could either: (1) pop up a menu with items "Set Time Limit", "Re-open", and "Close", or (2) open a single dialog with buttons to adjust the time or re-open/close. I have a slight preference for immediately opening a single dialog, if it can be made to look ok.

If we keep the menu items, can we change "Set Time Limit" to "Change Time Limit"? I was surprised when I clicked "Set Time Limit" but the actual dialog defaulted to "Add". I think that defaulting to "Add" (when there is an existing time limit) is the right behavior, but the menu item being called "Set" made me think it was going to default to "Setting" the time limit.

Reply:
Like this?
![image](https://user-images.githubusercontent.com/1004907/103470096-8c1efd00-4d22-11eb-8c5e-3166740a1b78.png)

Reply:
I kept the Delete/Regrade/Re-open/Close in the Actions menu (where they were before), but moved the Change action to a button in the Remaining column. This button is also now Enabled when the setting is "Closed", in that case the exam is re-opened and the time limit is set as needed.